### DISCUSSION TITLE
The go project uses skywalking-go and cannot be concatenated with Java projects

### DISCUSSION THREAD

**Discussion by InYourFuture** [apache/skywalking#11587] [Q&A]
The go service uses skywalking-go and send a request to java service,i can't find only traceid in skywalking ui,there are two traceid in skywalking. go service cannot be concatenated with Java service?

**Comment by mrproliu** [comment]
They should share the same trace ID, It's already [verified through the E2E in the main repo](https://github.com/apache/skywalking/blob/master/test/e2e-v2/cases/go/expected/trace-correlation-detail.yml). I think you should give more context in there? such as some trace, and topology screenshots?

**Reply by InYourFuture** [reply]
skywalking-backend:v8.9.1
skywalking-agent：v8.9.0
skywalking-go:v0.3.0
java service request go service. and java server service ip address is http://10.112.96.198:8088/hello
java service name:skywalking-java-server-local
goservice name:skywalking-demo-client
![image](https://github.com/apache/skywalking/assets/51349569/0d90d03c-9ec0-4061-8593-caf8e76f5be4)
![image](https://github.com/apache/skywalking/assets/51349569/b5a71ec3-1189-47d7-a009-d2315ebe7bef)
skywalking-java-server-local:
![image](https://github.com/apache/skywalking/assets/51349569/4dd8dcf7-3135-4c4d-959e-815953698a58)
![image](https://github.com/apache/skywalking/assets/51349569/caab7110-1e5f-4265-b5d2-35b7c1b7cca7)

skywalking-demo-client:
![image](https://github.com/apache/skywalking/assets/51349569/6c303f69-31e2-4866-95ee-b018a8d45873)
![image](https://github.com/apache/skywalking/assets/51349569/8e4c9d89-74f1-4ccc-8212-bd593395c890)
![image](https://github.com/apache/skywalking/assets/51349569/e2e02b47-2dcb-4813-ac2f-ee0f015ac6af)

**Reply by mrproliu** [reply]
I don't understand your system, you can determine if the trace is inconsistent through some means. For example, by printing trace-related logs in Java and Go programs, if there is still inconsistency here, I think you may need to locate whether the header information sent from the Go program to the Java service exists.

**Reply by InYourFuture** [reply]
but the go program request another go program,the topology is right and use asme traceid
The same code, that's why I'm confused
![image](https://github.com/apache/skywalking/assets/51349569/5b6f2291-96da-4a6a-9747-f85069ebc9de)

**Reply by mrproliu** [reply]
So I want you to take a deep look at two different language agents. Please note that, as I mentioned before, we have already verified the compatibility of golang and java languages in the [end-to-end testing of the main repo](https://github.com/apache/skywalking/blob/master/test/e2e-v2/cases/go/expected/trace-correlation-detail.yml). You can try it out following the method I recommended first.

**Reply by wu-sheng** [reply]
Try latest Java agent and backend, and the Java plugin relative with your case.

Your choosen versionz are super super old. Years ago.

**Reply by InYourFuture** [reply]
i try skywalking-java-agent-v9.0.0 is work ,thanks for your reply
